packages/peregrine/lib/store/actions/checkout/asyncActions.js

```diff
@@ -113,35 +95,41 @@ export const getShippingMethods = () => {
 
 export const submitPaymentMethodAndBillingAddress = payload =>
     async function thunk(dispatch, getState) {
-        submitBillingAddress(payload.formValues.billingAddress)(
-            dispatch,
-            getState
-        );
-        submitPaymentMethod(payload.formValues.paymentMethod)(
-            dispatch,
-            getState
-        );
+        const { countries, formValues } = payload;
+        const { billingAddress, paymentMethod } = formValues;
+
+        submitBillingAddress({
+            billingAddress,
+            countries
+        })(dispatch, getState);
+
+        submitPaymentMethod(paymentMethod)(dispatch, getState);
     };
 
 export const submitBillingAddress = payload =>
     async function thunk(dispatch, getState) {
         dispatch(actions.billingAddress.submit());
 
-        const { cart, checkout } = getState();
-        const { countries } = checkout;
+        const { cart } = getState();
```

Comment:
Nice. Each time we get rid of some slice of state that is used within an action God creates a kitten.